src/subdomains/core/history/controllers/history.controller.ts

```diff
@@ -68,14 +71,26 @@ export class HistoryController {
     @Headers('DFX-ACCESS-SIGN') sign: string,
     @Headers('DFX-ACCESS-TIMESTAMP') timestamp: string,
   ): Promise<CoinTrackingApiHistoryDto[]> {
-    const user = await this.userService.checkApiKey(key, sign, timestamp);
+    const user =
+      key.charAt(key.length - 1) === '0'
+        ? await this.userService.checkApiKey(key, sign, timestamp)
+        : await this.userDataService.checkApiKey(key, sign, timestamp);
     query = Object.assign(query, ApiKeyService.getFilter(user.apiFilterCT));
 
     return (
-      await this.historyService.getJsonHistory(
-        { format: ExportFormat.JSON, userAddress: user.address, ...query },
-        ExportType.COIN_TRACKING,
-      )
+      user instanceof User
+        ? await this.historyService.getJsonHistory(
+            { format: ExportFormat.JSON, userAddress: user.address, ...query },
+            ExportType.COIN_TRACKING,
+          )
+        : await Promise.all(
+            user.users.map(async (u) => {
+              return await this.historyService.getJsonHistory(
+                { format: ExportFormat.JSON, userAddress: u.address, ...query },
+                ExportType.COIN_TRACKING,
+              );
+            }),
+          ).then((t) => t.flat())
```

Comment:
Can't we use the new transaction history thing instead of using the Promise.all (which is probably more expensive)?

Reply:
I think I tried it, but will take a look in it again 🤔